### PR TITLE
task/common-types

### PR DESCRIPTION
**Pull request by shishnk**

### CODE REVIEW

File: Shared.Abstract/Primitives/Entity.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
+﻿namespace TaskNotFound.Shared.Abstract.Primitives;
+
+public abstract class Entity<TId>(TId id) : IEquatable<Entity<TId>>
+    where TId : notnull
+{
+    public TId Id { get; } = id;
+
+    public bool Equals(Entity<TId>? other)
+    {
+        if (ReferenceEquals(null, other))
```
Comment (makeitvsolo) [inline Shared.Abstract/Primitives/Entity.cs:10]: you can just `if (other is not null)`

File: Shared.Abstract/Primitives/Entity.cs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        {
+            return true;
+        }
+
+        return obj is Entity<TId> entity && Equals(entity);
+    }
+
+    public override int GetHashCode() => Id.GetHashCode();
+
+    public static bool operator ==(Entity<TId>? left, Entity<TId>? right) => Equals(left, right);
```
Comment (makeitvsolo) [inline Shared.Abstract/Primitives/Entity.cs:35]: Is it worth overloading operators at all?

File: Shared.Abstract/Primitives/Entity.cs (hunk 1 above)
Comment (shishnk) [reply Shared.Abstract/Primitives/Entity.cs:10]: These are completely identical expressions

File: Shared.Abstract/Primitives/Entity.cs (hunk 2 above)
Comment (shishnk) [reply Shared.Abstract/Primitives/Entity.cs:35]: == compares objects references

File: Shared.Abstract/Primitives/Entity.cs (hunk 1 above)
Comment (makeitvsolo) [reply Shared.Abstract/Primitives/Entity.cs:10]: But my variant already merged by other PRs

File: Shared.Abstract/Primitives/Entity.cs (hunk 2 above)
Comment (makeitvsolo) [reply Shared.Abstract/Primitives/Entity.cs:35]: That's right

**Review by makeitvsolo — APPROVED** [review]

**Pull request merged**